bundles/statistics/statsgrid2016/plugin/TogglePlugin.js

```diff
@@ -6,75 +12,79 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.TogglePlugin', function (flyout
     this._clazz = 'Oskari.statistics.statsgrid.TogglePlugin';
     this._index = 4;
     this._defaultLocation = 'bottom right';
+    this._fixedLocation = false;
 
     this.flyoutManager.on('show', function (tool) {
         me.toggleTool(tool, true);
     });
     this.flyoutManager.on('hide', function (tool) {
         me.toggleTool(tool, false);
     });
-    this._fixedLocation = true;
+    this.handler = new PluginHandler(() => this.renderButtons());
 }, {
-    _setLayerToolsEditModeImpl: function () {
-        if (!this.element) {
-            return;
-        }
-        const tools = this.element.find('div');
-        if (this.inLayerToolsEditMode()) {
-            tools.addClass('tool-drag-disabled');
-        } else {
-            tools.removeClass('tool-drag-disabled');
-        }
-    },
+    _setLayerToolsEditModeImpl: function () {},
     getElement: function () {
         return this.element;
     },
+    hasUI: function () {
+        this.changeToolStyle();
```

Comment:
hasUI() should return a boolean value. Publisher uses this to check if the plugin should be shutdown when entering publisher.